python/ray/serve/application.py

```diff
@@ -0,0 +1,376 @@
+                runtime_env_updates
+            )
+
+            for deployment in self._deployments.values():
+                self._configure_runtime_env(deployment, runtime_env_updates)
+            self.deploy()
+
+            logger.info("\nDeployed successfully!\n")
+
+            while True:
+                statuses = serve_application_status_to_schema(self.get_statuses()).json(
+                    indent=4
+                )
+                logger.info(f"{statuses}")
+                time.sleep(10)
+
+        except KeyboardInterrupt:
+            logger.info("Got SIGINT (KeyboardInterrupt). Removing deployments.")
+            for deployment in self._deployments.values():
+                deployment.delete()
+            if len(serve.list_deployments()) == 0:
+                logger.info("No deployments left. Shutting down Serve.")
+                serve.shutdown()
+            sys.exit()
+
+    def to_json(self, f: Optional[TextIO] = None) -> str:
+        """
+        Returns this Application's deployments as a JSON string. Optionally
+        writes the JSON string to a file as well. To write to a file, use this
+        pattern:
+
+        with open("file_name.txt", "w") as f:
+            app.to_json(f=f)
+
+        This JSON adheres to the Serve REST API schema. It can be deployed
+        via the Serve REST API.
+
+        Args:
+            f (Optional[TextIO]): A pointer to the file where the JSON should
+                be written.
+
+        Returns:
+            String: The deployments' JSON string. The output is similar to
+                json.dumps().
+        """
+
+        json_str = serve_application_to_schema(self._deployments).json(indent=4)
+
+        if f:
+            f.write(json_str)
+        return json_str
+
+    @classmethod
+    def from_json(cls, str_or_file: Union[str, TextIO]) -> "Application":
+        """
+        Takes in a string or a file pointer to a file containing deployment
+        definitions in JSON. These definitions are converted to a new
+        Application object containing the deployments.
+
+        To read from a file, use the following pattern:
+
+        with open("file_name.txt", "w") as f:
+            app = app.from_json(str_or_file)
+
+        Args:
+            str_or_file (Union[String, TextIO]): Either a string containing
+                JSON deployment definitions or a pointer to a file containing
+                JSON deployment definitions. The JSON format must adhere to the
+                ServeApplicationSchema JSON Schema defined in
+                ray.dashboard.modules.serve.schema.
+
+        Returns:
+            Application: a new Application object containing the deployments.
+        """
+
+        if isinstance(str_or_file, str):
+            schema = ServeApplicationSchema.parse_raw(
+                str_or_file, content_type="application/json"
+            )
+        else:
```

Comment:
any reason not to unify this and use `json.loads` in this case?

Reply:
+1, `from_json` is ambiguous, can I suggest the following?
- `from_dict` 
- `to_dict`
- `to_yaml` (for convenience)

Reply:
In this way we don't have a lot of I/O converter methods